Lockfileless / Dependency graphs (#1870)
semgrep/semgrep-docs: docs/semgrep-supply-chain/getting-started.md

 </tr>
</table>

## Scan a project without lockfiles (beta)

:::info

> Semgrep does not support managed scans of projects without lockfiles.

Reviewer: Are we saying that SMS doesn't scan projects without lockfiles at all? What happens? The SSC scan fails? Is there a way for the user to receive feedback? I feel like this is going to trip up users.

Author: @bkettle can you please clarify? ☝️

Reviewer 2: Ah, good point, I could see how this could be confusing. All that happens is that we aren't able to determine the dependencies when there is no lockfile, so the supply chain scan will not find any findings (but nothing will fail).

> :::

Reviewer: Would also suggest adding a header or separate section explaining to people how to enable this feature. "To participate in this beta, reach out to sales@semgrep.com", something like that.

Author: @bkettle I thought only Dependency Search requires action on our end; does lockfileless require us to enable something, too?

Reviewer 2: Lockfileless only requires using the `--enable-local-builds` flag, there is no feature flag on our side

> Semgrep supports the scanning of Java projects **without lockfiles** if they're built using Maven or Gradle with the help of the Gradle Wrapper. To do so:

Reviewer: ```suggestion
Semgrep Supply Chain can scan Java projects built using Maven or Gradle Wrapper without the need for lockfiles.
```

Just a suggestion, trying to improve clarity.

> 1. Ensure that the environment where you are initiating the Semgrep scan has all of the dependencies required to build your project, such as Java and Maven, installed.

Reviewer: ```suggestion
1. Ensure that the environment where you run Semgrep scans has installed all of the dependencies required to build your project, such as Java and Maven.
```

2. Initiate a Semgrep scan, ensuring that you include the `--allow-local-builds` flag:
    ```console
    semgrep ci --allow-local-builds

>     ```

Reviewer: ```suggestion
    ```
    For existing CI jobs, you may have to edit your configuration file to include this flag.
```
